sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -272,57 +333,318 @@ void testAssociateNewLocationForHandle() {
+                                (ManipulationPreconditions.AssociateLocationsManipulation
+                                                .EpisodicContextReportStateHandleObserver)
+                                        observer.get();
+                    }
+                }
+                observers.add(mdibAccessObserver);
+                final ContextStateModificationMessage report = new ContextStateModificationMessage(
+                        mockMdibAccess, Map.of("foo", List.of(mockLocationContextState, mockLocationContextState2)));
+                mdibAccessObserver.onUpdate(report);
+
+                observer.set(null);
+            }
+        });
+        t.start();
+
         assertFalse(
                 ManipulationPreconditions.AssociateLocationsManipulation.manipulation(injector),
                 "manipulation should've failed.");
         assertTrue(testRunObserver.isInvalid(), "Test run should have been invalid.");
+
+        for (MdibAccessObserver obs : observers) {
+            verify(mockMdibAccessObservable).registerObserver(obs);
+        }
     }
 
     @Test
     @DisplayName("associateNewLocations: Already existent state")
     void testAssociateNewLocationForHandleUsedState() {
         associateNewLocationsSetup();
+        final List<MdibAccessObserver> observers = new ArrayList<>();
 
         // introduce error, first state handle already in entity
         when(mockEntity.getStates(LocationContextState.class)).thenReturn(List.of(mockLocationContextState));
+
+        final MdibAccessObservable mockMdibAccessObservable = mock(MdibAccessObservable.class);
+        when(mockTestClient.getSdcRemoteDevice().getMdibAccessObservable()).thenReturn(mockMdibAccessObservable);
+        final AtomicReference<MdibAccessObserver> observer = new AtomicReference<>();
+        final Object signal = new Object();
+        doAnswer((arguments) -> {
+                    synchronized (signal) {
+                        observer.set(arguments.getArgument(0));
+                        signal.notifyAll();
+                    }
+                    return null;
+                })
+                .when(mockMdibAccessObservable)
+                .registerObserver(any());
+
+        final Thread t = new Thread(() -> {
+            ManipulationPreconditions.AssociateLocationsManipulation.EpisodicContextReportStateHandleObserver
+                    mdibAccessObserver;
+            for (int i = 0; i < 2; i++) {
+                mdibAccessObserver =
+                        (ManipulationPreconditions.AssociateLocationsManipulation
+                                        .EpisodicContextReportStateHandleObserver)
+                                observer.get();
+                synchronized (signal) {
+                    while (mdibAccessObserver == null) {
+                        try {
+                            signal.wait(300);
+                        } catch (InterruptedException e) {
```

Comment:
do not catch it here if you are not reacting to it

also you are not even interrupting the thread from another thread, thus there is no good way of reacting to it and it should just be escalated

Reply:
Improved just like above.